test-core/src/main/java/io/micronaut/test/annotation/MicronautTest.java

```diff
@@ -74,6 +74,12 @@
      */
     boolean transactional() default true;
 
+    /**
+     * Whether to wrap the setup and tests in a single transaction instead of using two separate transactions.
+     * @return true if the setup and tests should be wrapped in a single transaction
+     */
+    boolean singleTransaction() default false;
```

Comment:
I think I prefer an enum for this called `transactionMode` with an enum value that describes and documents the behaviour because otherwise we are going to get someone else asking for a different transactional behaviour and then we will have to add yet another member to MicronautTest

Reply:
Done. Please, let me know if the name of the enum constants and the documentation are OK. Feel free to make any improvements you see fit.